k8s/plugin/src/main.rs

```diff
@@ -174,13 +174,15 @@ async fn execute(cli_args: CliArgs) {
                 .map_err(|_e| {
                     std::process::exit(1);
                 });
-                resources
-                    .apply(
-                        &cli_args.namespace,
-                        cli_args.kube_config_path,
-                        cli_args.timeout,
-                    )
-                    .await;
+                if !resources.dry_run {
```

Comment:
Would we want to dry_run the apply as well?
Example: show which operations we would do on apply?

Reply:
Will add that in another PR , For now merging this PR